src/main/java/eu/darkbot/api/future/FutureResult.java

```diff
@@ -0,0 +1,60 @@
+package eu.darkbot.api.future;
+
+import java.util.Optional;
+import java.util.function.Consumer;
+
+public interface FutureResult<R> {
+
+    /**
+     * @return {@link Optional#empty()} if {@link FutureResult#isDone()} is false
+     */
+    Optional<R> getResult();
+
+    /**
+     * @return current {@link Status} of this {@link FutureResult}
+     */
+    Status getStatus();
+
+    /**
+     * May return false if current status is already {@link Status#COMMITTED}
+     * and mayInterrupt parameter is set to false.
```

Comment:
ok sure, but what does it do?

Does it have side-effects?
Does it mark task as completed?
Will it result in onDone being called?

The javadocs on this class are a bit lacking for how important of a class it is, document all the states in Status rigurously